File: assemblyadapter-core/src/main/java/com/github/panpf/assemblyadapter/OnLongClickListener.java
package com.github.panpf.assemblyadapter;

import android.content.Context;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public interface OnLongClickListener<DATA> {
    boolean onViewLongClick(@NonNull Context context, @NonNull View view, int position, @Nullable DATA data);
}

File: assemblyadapter-common-item/src/main/java/com/github/panpf/assemblyadapter/OnLongClickListener.kt
package com.github.panpf.assemblyadapter

import android.content.Context
import android.view.View

fun interface OnLongClickListener<DATA> {
    fun onLongClick(
        context: Context,
        view: View,
        bindingAdapterPosition: Int,
        absoluteAdapterPosition: Int,
        data: DATA
    ): Boolean
}